homeassistant/components/netatmo/climate.py

```diff
@@ -124,6 +130,14 @@ def _create_entity(netatmo_device: NetatmoRoom) -> None:
         {vol.Required(ATTR_SCHEDULE_NAME): cv.string},
         "_async_service_set_schedule",
     )
+    platform.async_register_entity_service(
+        SERVICE_SET_PRESET_MODE_WITH_OPTIONAL_END_DATETIME,
+        {
+            vol.Required(ATTR_PRESET_MODE): cv.string,
```

Comment:
We can check already here in the schema with `vol.In`

Reply:
Great catch. Thanks.